api/src/main/java/com/epam/pipeline/manager/pipeline/RunScheduleManager.java

```diff
@@ -135,73 +148,115 @@ public void init() {
-    public void deleteRunSchedulesForRun(final Long runId) {
-        runScheduleDao.deleteRunSchedulesForRun(runId);
+    public void deleteSchedules(final Long schedulableId, final ScheduleType scheduleType) {
+        runScheduleDao.deleteRunSchedules(schedulableId, scheduleType);
     }
 
-    public RunSchedule loadRunSchedule(final Long id) {
+    public RunSchedule loadSchedule(final Long id) {
         return runScheduleDao.loadRunSchedule(id).orElseThrow(() -> new IllegalArgumentException(
             messageHelper.getMessage(MessageConstants.ERROR_RUN_SCHEDULE_NOT_FOUND, id)));
     }
 
-    private void checkNewScheduleRequirements(final Long runId, final PipelineRun pipelineRun,
-                                              final PipelineRunScheduleVO runScheduleVO) {
-        checkIdenticalCronExpressionForRun(runId, runScheduleVO);
-        verifyRunSchedule(runId, pipelineRun, runScheduleVO);
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void deleteSchedulesForRunByPipeline(final Long pipelineId) {
+        runScheduleDao.deleteRunSchedulesForRunByPipeline(pipelineId);
+    }
+
+    private void checkNewRunScheduleRequirements(final Long schedulableId, final PipelineRunScheduleVO runScheduleVO,
+                                                 final ScheduleType scheduleType) {
+        checkIdenticalCronExpression(schedulableId, scheduleType, runScheduleVO);
+        verifyRunSchedule(schedulableId, runScheduleVO, scheduleType);
     }
 
-    private void verifyRunSchedule(final Long runId, final PipelineRun pipelineRun,
-                                   final PipelineRunScheduleVO runScheduleVO) {
+    private void verifyRunSchedule(final Long runId, final PipelineRunScheduleVO runScheduleVO,
+                                   final ScheduleType type) {
         verifyCronExpression(runId, runScheduleVO);
         Assert.notNull(runScheduleVO.getAction(),
                        messageHelper.getMessage(MessageConstants.SCHEDULE_ACTION_IS_NOT_PROVIDED, runId));
-        Assert.notNull(pipelineRun,
-                       messageHelper.getMessage(MessageConstants.ERROR_RUN_PIPELINES_NOT_FOUND, pipelineRun.getName()));
-        Assert.isTrue(!pipelineRun.getInstance().getSpot(), messageHelper.getMessage(
-            MessageConstants.ERROR_ON_DEMAND_REQUIRED));
-        Assert.isTrue(!pipelineRun.getStatus().isFinal(), messageHelper.getMessage(
-            MessageConstants.ERROR_PIPELINE_RUN_FINISHED, runId));
         Assert.isTrue(StringUtils.hasText(runScheduleVO.getTimeZone()),
                       messageHelper.getMessage(MessageConstants.ERROR_TIME_ZONE_IS_NOT_PROVIDED, runId));
-        Assert.isTrue(!(runScheduleVO.getAction().equals(RunScheduledAction.PAUSE)
-                        && isNonPauseOrClusterRun(pipelineRun)),
-                      messageHelper.getMessage(MessageConstants.DEBUG_RUN_IDLE_SKIP_CHECK));
+        switch (type) {
+            case PIPELINE_RUN:
```

Comment:
I'd suggest replacing `switch` with interface call where implementation is selected depending on type, similar to storage handling approach.